src/components/project/dto/step.enum.ts

```diff
@@ -61,4 +61,10 @@ export abstract class ProjectStepTransition {
 
   @Field(() => TransitionType)
   type: TransitionType;
+
+  @Field(() => Boolean, { defaultValue: false })
+  disabled?: boolean;
+
+  @Field(() => String, { defaultValue: '' })
+  disabledLabel?: string;
```

Comment:
Maybe `disabledReason` would be clearer? This should be nullable too. Empty strings should not be used to represent a lack of a value.

Reply:
Hmm. Why should this be nullable, but not `disabled`? It seems like every time `disabled` is true `disabledReason` would exist.

Reply:
I guess then you can have consistent boolean checks for `disabled`.